diff --git a/kubeval/kubeval.go b/kubeval/kubeval.go
index 10e0eef..40f9cd7 100644
--- a/kubeval/kubeval.go
+++ b/kubeval/kubeval.go
@@ -73,9 +73,7 @@ func determineSchemaURL(baseURL, kind, apiVersion string, config *Config) string
 	}
 
 	groupParts := strings.Split(apiVersion, "/")
-	versionParts := strings.Split(groupParts[0], ".")
-
-	kindSuffix := "-" + strings.ToLower(versionParts[0])
+	kindSuffix := "-" + strings.ToLower(groupParts[0])
 	if len(groupParts) > 1 {
 		kindSuffix += "-" + strings.ToLower(groupParts[1])
 	}
